Lib/asyncio/taskgroups.py

```diff
@@ -115,10 +115,11 @@ async def __aexit__(self, et, exc, tb):
         if self._base_error is not None:
             raise self._base_error
 
-        if propagate_cancellation_error is not None:
-            # The wrapping task was cancelled; since we're done with
-            # closing all child tasks, just propagate the cancellation
-            # request now.
+        # Propagate CancelledError if there is one, except if there
+        # are other errors and we're not required by uncancel() != 0.
+        # (The latter means we *must* propagate the cancellation.)
+        if (propagate_cancellation_error is not None and
```

Comment:
Doesn't this now suppress the TimeoutError when used with `async with timeout()`?

Reply:
Since the condition is true in fewer cases than the original code (where it was just `if propagate_cancellation_error is not None:`) I don't see how it could suppress something that wouldn't be suppressed previously. But I'm not sure what you're thinking of, so please make a repro.